pkg/recipes/terraform/config/config.go

```diff
@@ -136,12 +136,29 @@ func newModuleConfig(moduleSource string, moduleVersion string, params ...Recipe
 }
 
 // getProviderConfigs generates the Terraform provider configurations for the required providers.
-func getProviderConfigs(ctx context.Context, requiredProviders []string, supportedProviders map[string]providers.Provider, envConfig *recipes.Configuration) (map[string]any, error) {
-	providerConfigs := make(map[string]any)
+func getProviderConfigs(ctx context.Context, requiredProviders []string, ucpSupportedProviders map[string]providers.Provider, envConfig *recipes.Configuration) (map[string][]any, error) {
+	providerConfigs := make(map[string][]any)
+
+	// Get recipe provider configurations from the environment configuration
+	recipeProviderConfigs := getRecipeProviderConfigs(ctx, envConfig)
+
+	// Add provider config from recipeProviderConfigs to providerConfigs
+	for provider, config := range recipeProviderConfigs {
+		if len(config) > 0 {
+			providerConfigs[provider] = config
+		}
+	}
+
+	// Build provider configurations for required providers excluding the ones already present in providerConfigs
```

Comment:
Could you remind me the user experience we decided on again - unlike the current requirement, a user doesn't *need* to specify providers used in the module in the required providers anymore?

The reason I'm asking is because this code will add all the providers configured on the environment to the terraform config even if they are not used by the module. We should validate that this doesn't cause unexpected behavior.

Reply:
Currently we are adding to main.tf.json all providers configured in the envConfig/RecipeConfig/Terraform/Providers section and adding any additional configurations from the required_provider section the user may have configured in recipes. Let's discuss if this is not expected.